projects/dex-ui/src/utils/what-changed.ts

```diff
@@ -0,0 +1,193 @@
+type AnyObject = Array<any> | Record<string, any>;
+type Diffs = Array<{
+  path: string;
+  oldValue: any;
+  newValue: any;
+}>;
+
+/** is pojo or array */
+function isObj(obj: any) {
+  return typeof obj === "object" && obj !== null;
+}
+
+/**
+ * compares a react dependencies list with its previous values (from previous cycle)
+ * and prints which dependency is the one that changed
+ */
+export const reactWhatChanged = function reactWhatChanged(dependencies: AnyObject, verbose: boolean = false, id?: string | number) {
```

Comment:
🤔  What's all this about? Looks like we aren't using this?

Reply:
This is a useful function to spit out what changed in a dependency array. Was trying to figure out what was causing extra renders, and this showed me.

```
useEffect(() => {
  // use one, two
}, reactWhatChanged([one, two]))
```